letter-shop.md

```diff
@@ -0,0 +1,190 @@
+
+### Prepare and Fulfill with on-ledger Escrow
+```js
+function base64url (buf) { return buf.toString('base64').replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '') }
+
+// [...]
+
+const secret = crypto.randomBytes(32)
+const fulfillment = base64url(secret)
+const condition = base64url(crypto.createHash('sha256').update(secret).digest())
+```
+
+The plugin used here is specific to XRP, and to the use of on-ledger escrow. Escrow for XRP is described [here](https://ripple.com/build/rippleapi/#transaction-types). Escrow transfers differ from normal transfer in that the recipient doesn't automatically receive the amount of the transfer in their account; the need to produce something in order to claim the funds. During the time between the sender's action of preparing the transfer (creating the escrow), and the time the recipient produces the fulfillment for the transfer's condition, the money is on hold on the ledger. If the recipient doesn't produce the fulfillment in time, the transaction is canceled, and the money goes back into the sender's account.
+
+### Hashlocks
+In the case of Interledger, the fulfillment is always a 32-byte string, and the condition is the sha256 hash of that string.
+SHA256 is a one-way hash function, so if you know `fulfillment`, then it's easy and quick to calculate `condition = sha256(fulfillment)`, but if you only know the condition, if you only have a million years or less, it's near-impossible to find (i.e., guess) a fulfillment for which `condition = sha256(fulfillment)` would hold. And in practice, we tend to use rollback timeouts that are of course much shorter! :) Because the transfer is *locked* until the recipient produces the correct fulfillment, the condition of the transfer is a *hash* of its fulfillment, we call this a *hashlock*.
+
+## Step 2: Paying for your letter
+Visit http://localhost:8000. You'll see something like:
+```txt
+Please send an Interledger payment by running: node ./pay.js test.crypto.xrp.rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW 10 nhPJyYh-KkZSMHz8dfOQZAmCRAGnO39b0iFwV5qOmOA
+```
+
+Follow these instructions and wait for about 30 seconds, until you see something like:
+```txt
+Got the fulfillment, you paid for your letter! Go get it at http://localhost:8000/RWtoGF_sOVoIH3Casd-nmApQ-Thzl03lH-cInXume_g
+```
+
+As the instructions say, visit that URL to get your letter! :)
+
+### Sending transfers
+
+To send an Interledger payment, call the `sendTransfer` method of any connected Interledger plugin:
+
+```js
+function sendTransfer (obj) {
```

Comment:
I think it's more confusing for there to be a separate function called sendTransfer. Why not just put these parameters into the call below?